modules/decline/src/Smithy4sCli.scala

```diff
@@ -128,7 +120,17 @@ class Smithy4sCli[Alg[_[_, _, _, _, _]], Op[_, _, _, _, _]](
       }
   }
 
-  def main: Opts[IO[ExitCode]] = service.endpoints
+  private def opts: Opts[IO[ExitCode]] = service.endpoints
     .foldMapK(endpointSubcommand(_))
 
+  def command: Command[IO[ExitCode]] = {
```

Comment:
if we're getting rid of the `CommandIOApp` thing, couldn't this be polymorphic ?

Reply:
yes I believe so